### PR TITLE
Add support for PHP 8.0

### PR DESCRIPTION
**Pull request by pamil**
Part of #12298.

<hr />

**TODO:**
 - [x] Better userland deprecations handling in phpspec

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -190,7 +190,6 @@
         "phpunit/phpunit": "^8.5",
         "psalm/plugin-mockery": "0.7.0",
         "psr/event-dispatcher": "^1.0",
-        "sspooky13/yaml-standards": "^6.0",
```
Comment (pamil) [inline composer.json:193]: https://github.com/sspooky13/yaml-standards/issues/56

File: .github/workflows/application.yml [hunk 2]
```diff
@@ -119,14 +119,10 @@ jobs:
...
             fail-fast: false
             matrix:
-                php: [7.4]
-                symfony: [^4.4, ^5.2]
-                mysql: [5.7, 8.0]
-
-                exclude:
-                    -
-                        php: 7.3
-                        mysql: 8.0
```
Comment (pamil) [inline .github/workflows/application.yml:129]: PHP 7.3 is long gone.

File: phpspec.yml.dist [hunk 3]
```diff
@@ -42,3 +42,5 @@ suites:
     UserBundle: { namespace: Sylius\Bundle\UserBundle, psr4_prefix: Sylius\Bundle\UserBundle, spec_path: src/Sylius/Bundle/UserBundle, src_path: src/Sylius/Bundle/UserBundle }
 
     Behat: { namespace: Sylius\Behat, psr4_prefix: Sylius\Behat, spec_path: src/Sylius/Behat, src_path: src/Sylius/Behat }
+
+runner.maintainers.errors.level: 14335 # E_ALL & ~E_USER_DEPRECATED ^ E_STRICT
```
Comment (pamil) [inline phpspec.yml.dist:46]: Will be contributed back to phpspec soon.

File: src/Sylius/Bundle/CoreBundle/composer.json [hunk 4]
```diff
@@ -26,10 +26,10 @@
         }
     ],
     "require": {
-        "php": "^7.4",
+        "php": "^7.4 || ^8.0",
         "doctrine/doctrine-migrations-bundle": "^3.0",
         "egulias/email-validator": "^3.0",
-        "fzaninotto/faker": "^1.9",
+        "fakerphp/faker": "^1.10",
```
Comment (pamil) [inline src/Sylius/Bundle/CoreBundle/composer.json:32]: It has been changed in main Sylius/Sylius dependencies already.

**Comment by loevgaard** [comment]
❤️ ❤️ ❤️

**Review by lchrusciel — APPROVED** [review]

**Pull request merged**